src/streams.rs

```diff
@@ -110,15 +110,19 @@ where
                 let now = SystemTime::now();
                 let update = this
                     .last
-                    .and_then(|last| now.duration_since(last).ok().map(|d| d >= UPDATE_INTERVAL))
+                    .map(|last| now.duration_since(last).unwrap_or_default() >= UPDATE_INTERVAL)
```

Comment:
Shouldn't change the behavior, but makes it easier to understand the intent.
```suggestion
                    .map(|last| now.duration_since(last).unwrap_or(Duration::ZERO) >= UPDATE_INTERVAL)
```